### DISCUSSION TITLE
React Query with React Native

### DISCUSSION THREAD

**Discussion by shaneajeffery** [TanStack/query#916] [General]
Is it possible to use React Query with React Native the same way you would use it with React for web?

Given that Redux and Apollo Client can be leveraged literally the same way, I am thinking the answer is yes, but I want to make sure that I am not missing anything with your implementation of React Query that would make it not behave properly or non-optimal for mobile.

Thanks!

**Comment by Covicake** [comment]
I've been using it for a few weeks and everything does work except that when you have a cache that lives more than a few minutes (for example, 5 minutes) you get a warning about using long-lived timers being a performance and correctness problem. I've resorted to just set cacheTime to 0, because I don't really need cache

**Reply by tannerlinsley** [reply]
To get rid of these errors, you can swap out the timeout/interval logic in RN with something that won't warn. It comes with tradeoffs though.
